crates/solver/src/driver.rs

```diff
@@ -420,13 +445,36 @@ impl Driver {
                     .map_err(|err| anyhow!("{err}"))
                     .context("convert nonce")?,
             };
+            let scores = model::solver_competition::Scores {
+                winner: scores.peek().expect("no winner").0,
+                winning_score: scores.next().expect("no score").1, // guaranteed to exist
+                // reference score is the second highest score, or 0 if there is only one score (see
+                // CIP20)
+                reference_score: scores.next().unwrap_or_default().1,
+                block_deadline: {
+                    let deadline = self.solver_time_limit
+                        + self.solution_submitter.max_confirm_time
+                        + self.additional_mining_deadline;
+                    let number_blocks = deadline.as_secs() / self.block_time.as_secs();
+                    block_during_simulation + number_blocks
+                },
+            };
+            let participants = solver_competition
+                .solutions
+                .iter()
+                .map(|solution| solution.solver_address)
+                .collect::<HashSet<_>>() // to avoid duplicates
+                .into_iter()
+                .collect();
             tracing::debug!(?transaction, "winning solution transaction");
 
             let solver_competition = model::solver_competition::Request {
                 auction: auction_id,
                 transaction,
                 competition: solver_competition,
                 executions,
+                scores,
+                participants,
```

Comment:
Here specifically, we can add the relevant `auction_prices` to store. This is compatible with solver+driver colocation as this logic is moved to the `autopilot` anyway.

Reply:
Fixed here https://github.com/cowprotocol/services/pull/1282